apps/eth-wrapper/src/components/Wrapper.tsx

```diff
@@ -0,0 +1,127 @@
+                console.log(safeTx.safeTxHash);
+            } catch (e) {
+                console.error(e)
+            }
+        } else {
+            try {
+                const parsedAmount = ethers.utils.parseEther(amountToWrap);
+                const withdraw = new ethers.utils.Interface(WETHwithdraw_function);
+                const safeTx = await sdk.txs.send({
+                    txs: [{
+                        to: getWethAddress(safe.network.toLowerCase()),
+                        value: '0',
+                        data: withdraw.encodeFunctionData("withdraw", [parsedAmount])
+                    }]
+                })
+                setSafeTxHash(safeTx.safeTxHash);
+                console.log(safeTx.safeTxHash);
+            } catch (e) {
+                console.error(e);
+            }
+        }
+    }, [sdk, amountToWrap, isError, props.wrap, safe])
+
+    const validateAmout = useCallback((newValue: string) => {
+        console.log("available balance:", availableBalance);
+        console.log("new value: ", newValue)
+        if (isNaN(Number(newValue))) {
+            setIsError(true);
+            setErrorMessage("Not a number");
+        }
+        else if (Number.parseFloat(newValue) > availableBalance) {
+            setIsError(true);
+            setErrorMessage("Insufficient funds");
+        }
+        else {
+            setIsError(false);
+            setErrorMessage("");
+            setAmountToWrap(newValue);
+        }
+    }, [availableBalance])
+
+    useEffect(() => {
+        setAmountToWrap("");
+    }, [props.wrap])
+
+    useEffect(() => {
+        const runEffect = async () => {
+            // update available balance
+            var newValue = "0";
+            if (props.wrap) {
+                const balanceEth = await sdk.eth.getBalance([safe.safeAddress]);
+                newValue = ethers.utils.formatEther(balanceEth);
+            } else {
+                const balanceWeth = await weth.balanceOf(safe.safeAddress);
+                newValue = ethers.utils.formatEther(balanceWeth);
+            }
+            setAvailableBalance(Number.parseFloat(newValue));
+        };
+        runEffect();
+    }, [safe, sdk, weth, props.wrap]);
+
+    return (
+        <Grid container spacing={3}>
+            <Snackbar
+                open={safeTxHash.length !== 0}
+                autoHideDuration={3000}
+                onClose={() => setSafeTxHash("")}
+                message="You transaction has been submitted"
+            />
+            <Grid item xs={8}>
+                <TextField
+                    value={amountToWrap}
+                    label={props.wrap ? "ETH amount" : "WETH amount"}
+                    meta={{ error: errorMessage }}
+                    onChange={e => validateAmout(e.target.value)} />
+            </Grid>
+            <Grid item xs={8}>
+                <Button size="md" color="primary" variant="contained" onClick={() => wrapEth()}>
```

Comment:
Minor comment, but I see it in multiple parts: It is always called `wrapEth` which does not describe what it is actually doing.